Components/ATMega_PIN_Diagram.py

```diff
@@ -139,6 +139,8 @@ def __init__(self):
             self.temperatureFrame.layout = QVBoxLayout()
             self.temperatureFrame.layout.setAlignment(Qt.AlignLeft)
             self.temperatureFrame.layout.addStretch()
+            
+            self.verticalSlider.valueChanged[int].connect(self.changeValue)
 
 
             for val in temperature:
```

Comment:
What is the role of this loop?

Reply:
it detects if the value is changed then passes it on to the helper function.
thats how you fetch widget values in pyqt5
https://www.riverbankcomputing.com/static/Docs/PyQt5/signals_slots.html

Reply:
I was talking about this loop :   for val in temperature:

Reply:
I didn't write that code haha

Reply:
If there is no logic under for loop so can you just remove the for loop?